File: backend/src/main/java/com/jugglehive/model/entity/RaceSkill.java
package com.jugglehive.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Entity
@Data
public class RaceSkill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "race_id")
    private Race race;

    @ManyToOne
    @JoinColumn(name = "skill_tree_id")
    private TreeEntity tree;

    @Column(name = "slot")
    private Integer slot;
}

File: JuggleHiveWebapp.Server/Models/RaceSkill.cs
using System;
using System.Collections.Generic;

namespace JuggleHiveWebapp.Server.Models;

public partial class RaceSkill
{
    public long Id { get; set; }

    public long? RaceId { get; set; }

    public long? SkillTreeId { get; set; }

    public int? Slot { get; set; }

    public virtual Race? Race { get; set; }

    public virtual TreeEntity? SkillTree { get; set; }
}
